internal/groups/api/endpoint.go

```diff
@@ -0,0 +1,335 @@
+package groups
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/mainflux/mainflux/internal/groups"
+	"github.com/mainflux/mainflux/pkg/errors"
+)
+
+func CreateGroupEndpoint(svc groups.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(createGroupReq)
+		if err := req.validate(); err != nil {
+			return groupRes{}, err
+		}
+
+		group := groups.Group{
+			Name:        req.Name,
+			Description: req.Description,
+			ParentID:    req.ParentID,
+			Metadata:    req.Metadata,
+		}
+
+		created, err := svc.CreateGroup(ctx, req.token, group)
```

Comment:
`created` should be called `group` but in fact `svc.CreateGroup` should return `groupID` instead of whole Group.